src/player.js

```diff
@@ -1673,11 +1673,12 @@ export default class Player extends FakeEventTarget {
       this._eventManager.listen(this._engine, CustomEventType.TEXT_CUE_CHANGED, (event: FakeEvent) => this._onCueChange(event));
       this._eventManager.listen(this._engine, CustomEventType.ABR_MODE_CHANGED, (event: FakeEvent) => this.dispatchEvent(event));
       this._eventManager.listen(this._engine, CustomEventType.TIMED_METADATA, (event: FakeEvent) => this.dispatchEvent(event));
-      this._eventManager.listen(this._engine, CustomEventType.AUTOPLAY_FAILED, (event: FakeEvent) => {
+      this._eventManager.listen(this._engine, CustomEventType.PLAY_FAILED, (event: FakeEvent) => {
         this.pause();
         if (this._firstPlay && this._config.playback.autoplay) {
-          this.dispatchEvent(event);
+          this.dispatchEvent(new FakeEvent(CustomEventType.AUTOPLAY_FAILED));
```

Comment:
add the `PLAY_FAILED` error payload to the `AUTOPLAY_FAILED`.
If for some reason someone would want to get the details he would need to listen to both events which is an overhead.